### PR TITLE
Separate Releasing from "Latest" release determination

### LINKED ISSUE #47786: Decouple release from tags (latest/nightly/beta)
[discussion](https://metaboat.slack.com/archives/C864UT5CZ/p1723818227926829?thread_ts=1723794043.125729&cid=C864UT5CZ)

[discussion](https://metaboat.slack.com/archives/C0669P4AF9N/p1725418191282799)

- [x] remove `latest` tagging from publish action
- [x] Github action to set a tag on a version
- [x] update version-info logic
- [x] some sort of display of what is tagged how/where
- [ ] link to action in ops issues
- [ ] link nightly releases to latest tag

### PR DESCRIPTION
**Pull request by iethree**
Closes https://github.com/metabase/metabase/issues/47786

### Description

In a number of cases, we might not want the most recently released version be the official "latest" version. This change makes "latest" a fully manual determination.  (It also supports manually determining any number of other release channels: currently "latest", "beta" and "nightly" are included)

1. Removes the code that determines "latest" based on fixed logic.
2. Removes the code from the release process that uploads to the s3 `latest/` director, tags docker images with the `:latest` tag, and sets the "latest" property in version-info.json
3. Adds a new `release-tag` workflow that allows any version to be manually set as the "nightly", "beta", or "latest" by (a) uploading to that directory in s3, (2) setting that tag in dockerhub,(3) adding git tags for release channels, (4) for latest: setting the latest property in version-info.json. This workflow supports different channel designations for EE and OSS.
4. When the release-tag workflow sets a "latest" version, it adds a `rollout` property to the `latest` property in version-info.json that we can use in metabase instances to control what percentage of instances are notified of that upgrade.

Screen Shot 2024-09-11 at 3 29 33 PM

Screen Shot 2024-09-11 at 1 29 24 PM

5. Adds some updates to the release log workflow to show the channel tags in the release log, and adds a separate summary page which shows what versions are tagged with which release channels.

Screen Shot 2024-09-11 at 1 37 03 PM
Screen Shot 2024-09-11 at 1 36 52 PM

Note: this makes no in-product changes at all, and customer update notifications should only be affected insofar as this could change the timing of when a new "latest" version is show in the UI.

[Sample run in my fork
](https://github.com/iethree/metabase/actions/runs/10817852621)

To do after this merges:
- [ ] link this new action in ops issues
- [ ] automate `nightly` tag for auto patch releases

### CODE REVIEW

File: .github/workflows/release-tag.yml [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+        elif [[ "${{ matrix.edition }}" == "oss" ]]; then
+          git tag -f ${{ inputs.tag_name }}-oss ${{ needs.check-version.outputs.oss }}
+          git push origin -f ${{ inputs.tag_name }}-oss
+        fi
+
+  update-release-log:
+    needs: check-version
+    uses: ./.github/workflows/release-log.yml
+    with:
+      version: 50 # maybe should be vars.CURRENT_VERSION
```
Comment (crisptrutski) [inline .github/workflows/release-tag.yml:265]: Any reason not to do this yet?

File: release/src/release-log.ts [hunk 2]
```diff
@@ -21,9 +21,13 @@ export async function gitLog(majorVersion: number) {
 
 function processCommit(commitLine: string): CommitInfo {
   const [refs, message, hash, date] = commitLine.split('||');
-  const version = refs?.match(/(v[\d\.\-RCrc]+)/)?.[1] ?? '';
+  const tags = refs?.match(/tag: ([\w\d-_\.]+)/g) ?? '';
 
-  return { version, message, hash, date};
+  const versions = tags
+    ? tags?.map((v) => v.replace('tag: ', ''))
```
Comment (crisptrutski) [inline release/src/release-log.ts:27]: Why do we still need this conditional if we've checked `tags` is truthy in the ternion expression?

```suggestion
    ? tags.map((v) => v.replace('tag: ', ''))
```

File: release/src/version-info.unit.spec.ts [hunk 3]
```diff
@@ -39,29 +39,29 @@ describe("version-info", () => {
...
-    it("should add new latest version to version info json", () => {
+    it("should add new version to version info json", () => {
       const generatedJson = generateVersionInfoJson({
         milestoneIssues: issues,
         version: "v0.3.0",
         existingVersionInfo: oldJson,
       });
 
-      expect(generatedJson.latest).toEqual({
+      expect(generatedJson.older).toEqual([{
```
Comment (crisptrutski) [inline release/src/version-info.unit.spec.ts:49]: Given the way that versions are added to `older` before being marked as latest, the naming feels funny. How about using the key `other` instead?

**Review by crisptrutski — APPROVED** [review]
Decomplect all the things! 🧶

File: .github/workflows/release-tag.yml [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+      uses: actions/github-script@v7
+      id: canonical_version
+      with:
+        script: | # js
+          const { isValidVersionString, getCanonicalVersion, hasBeenReleased } = require('${{ github.workspace }}/release/dist/index.cjs');
+
+          const version = '${{ inputs.version }}';
+
+          if (!isValidVersionString(version)) {
+            throw new Error("The version format is invalid! It must start with either 'v0.' or 'v1.'.");
```
[the remark this answers is not in the document]
Comment (iethree) [reply .github/workflows/release-tag.yml:63]: this check is actually more than the just the prefix. I'll update the error message. 👍

File: .github/workflows/release-tag.yml (hunk 1 above)
Comment (iethree) [reply .github/workflows/release-tag.yml:265]: good point

File: release/src/version-info.unit.spec.ts (hunk 3 above)
Comment (iethree) [reply release/src/version-info.unit.spec.ts:49]: I agree, `other` would be a much better name, but I think we need to maintain the current name for the sake of backward compatibility with existing instances.

File: .github/workflows/release-tag.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (crisptrutski) [reply .github/workflows/release-tag.yml:63]: Yip was just referring to the message

**Pull request merged**